packages/builder/src/components/integration/QueryViewerSidePanel/index.svelte

```diff
@@ -41,7 +41,7 @@
   </div>
   <div class="content">
     {#if activeTab === "JSON"}
-      <JSONPanel data={rows[0] || {}} />
+      <JSONPanel data={rows?.length === 1 ? rows[0] : rows || {}} />
```

Comment:
Would it not be better to always return `rows` in this case instead of `rows[0]` if our intention is to show the full response, and not the first row?

Reply:
The issue is the way the responses are resolved. Most use cases are not actually arrays, but our code is parsing it to it. Doing this check we display the response as we did before in most of the cases, fe.
<img width="964" alt="image" src="https://github.com/Budibase/budibase/assets/15987277/8afb4948-1dcb-44cb-94a9-f774d0c55b5a">

Reply:
Gotcha - makes sense!